src/Symfony/Component/HttpClient/Response/ResponseTrait.php

```diff
@@ -147,21 +147,21 @@ public function toArray(bool $throw = true): array
         $contentType = $this->headers['content-type'][0] ?? 'application/json';
 
         if (!preg_match('/\bjson\b/i', $contentType)) {
-            throw new JsonException(sprintf('Response content-type is "%s" while a JSON-compatible one was expected.', $contentType));
+            throw new JsonException(sprintf('Response content-type is "%s" while a JSON-compatible one was expected for "%s".', $contentType, $this->getInfo('url')));
         }
 
         try {
             $content = json_decode($content, true, 512, JSON_BIGINT_AS_STRING | (\PHP_VERSION_ID >= 70300 ? JSON_THROW_ON_ERROR : 0));
         } catch (\JsonException $e) {
-            throw new JsonException($e->getMessage(), $e->getCode());
+            throw new JsonException(sprintf('JSON error: %s, from "%s".', $e->getMessage(), $this->getInfo('url')), $e->getCode());
```

Comment:
I'd suggest removing the prefix (same below):
`sprintf('%s for "%s".', $e->getMessage(), $this->getInfo('url')`